app/routes/account.$accountId.$appId.security/view.tsx

```diff
@@ -1,59 +1,102 @@
 import { Divider } from "@mantine/core"
+import { showNotification } from "@mantine/notifications"
 import { Box } from "@pokt-foundation/pocket-blocks"
-import { useMemo } from "react"
-import { BlockchainsQuery, PortalApp } from "~/models/portal/sdk"
+import { useFetcher } from "@remix-run/react"
+import { useReducer, useEffect } from "react"
+import { action, SecurityActionData } from "./route"
+import { DEFAULT_WHITELISTS, securityReducer } from "./utils/stateReducer"
+import {
+  BlockchainsQuery,
+  PortalApp,
+  WhitelistContractsV2,
+  WhitelistMethodsV2,
+} from "~/models/portal/sdk"
 import { Blockchain } from "~/models/portal/sdk"
 import ApprovedChains from "~/routes/account.$accountId.$appId.security/components/ApprovedChains"
 import ChainWhitelist from "~/routes/account.$accountId.$appId.security/components/ChainWhitelist"
 import PrivateSecretKey from "~/routes/account.$accountId.$appId.security/components/PrivateSecretKey"
 import WhitelistOrigins from "~/routes/account.$accountId.$appId.security/components/WhitelistOrigins"
 import WhitelistUserAgents from "~/routes/account.$accountId.$appId.security/components/WhitelistUserAgents"
-import {
-  formatBlockchainWhitelist,
-  WhitelistContract,
-  WhitelistMethod,
-} from "~/routes/account.$accountId.$appId.security/utils"
+import { LoaderDataStruct } from "~/utils/loader"
 
 type SecurityViewProps = {
+  actionData?: LoaderDataStruct<SecurityActionData>
   app: PortalApp
   blockchains: BlockchainsQuery["blockchains"]
 }
 
-export const SecurityView = ({ app, blockchains }: SecurityViewProps) => {
-  const whiteListContracts = useMemo(
-    () =>
-      formatBlockchainWhitelist<WhitelistContract>(app.whitelists.contracts, "contracts"),
-    [app.whitelists.contracts],
+export const SecurityView = ({ actionData, app, blockchains }: SecurityViewProps) => {
+  const [state, dispatch] = useReducer(
+    securityReducer,
+    app.whitelists ?? DEFAULT_WHITELISTS,
   )
+  const fetcher = useFetcher()
 
-  const whiteListMethods = useMemo(
-    () => formatBlockchainWhitelist<WhitelistMethod>(app.whitelists.methods, "methods"),
-    [app.whitelists.methods],
-  )
+  useEffect(() => {
+    console.log(state)
+    // stop it from posting on initial load
+    fetcher.submit(
```

Comment:
We still need to handle this?

Reply:
ah yeah, i forgot about that. probably should, but functionally its not noticeable